### PR TITLE
Delete Catalogue Solution

### PR DESCRIPTION
**Pull request by prakashpawarnhs**
Story [AB#9038](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/9038)
Task [AB#12972](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/12972), [AB#12973](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/12973)

### CODE REVIEW

**Comment by prakashpawarnhs** [comment]
Delete catalogue solutions and continue.

File: app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import manifest from './manifest.json';
+import { baseUrl } from '../../../../../../config';
+import { addParamsToManifest } from '../../../../../../helpers/contextCreators/addParamsToManifest';
+
+export const getContext = ({ orderId, solutionName }) => {
+  const context = ({
+    ...addParamsToManifest(manifest, { orderId, solutionName })
+  });
+  return context;
+};
```
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.js:10]: There are some linting issues with this file:

```
app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.js     
   2:10  error  'baseUrl' is defined but never used            no-unused-vars
   7:64  error  Missing trailing comma                         comma-dangle
  10:3   error  Newline required at end of file but not found  eol-last
```

File: app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.test.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      const context = getContext({});
+      expect(context.orderDescriptionTitle).toEqual(`${manifest.orderDescriptionTitle}`);
+    });
+
+    it('should return the description', () => {
+      const context = getContext({});
+      expect(context.description).toEqual(manifest.description);
+    });
+  });
+});
```
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.test.js:23]: There are a couple of linting issues with this file:

```shell
app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.test.js
  3:10  error  'baseUrl' is defined but never used           no-unused-vars
  7:1   error  Expected indentation of 4 spaces but found 6  indent
```

File: app/pages/sections/order-items/catalogue-solutions/delete/confirmation/controller.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  req,
+  sessionManager,
+  accessToken,
+  logger,
+}) => {
+  const { orderId } = req.params;
+  const { solutionName } = req.params;
+
+  return getContext({ orderId, solutionName });
+};
```
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/delete/confirmation/controller.js:13]: There are some linting issues in this file:

```shell
app/pages/sections/order-items/catalogue-solutions/delete/confirmation/controller.js
  5:3  error  'sessionManager' is defined but never used  no-unused-vars
  6:3  error  'accessToken' is defined but never used     no-unused-vars
  7:3  error  'logger' is defined but never used          no-unused-vars
```

File: app/pages/sections/order-items/catalogue-solutions/delete/confirmation/controller.test.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        req: { params: { orderId: 'order-item-id' } },
+        sessionManager: fakeSessionManager,
+        accessToken: 'access-token',
+        logger,
+      });
+
+      expect(actualContext).toEqual(expectedContext);
+    });
+  });
+});
```
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/delete/confirmation/controller.test.js:78]: There are some linting issues in this file:

```shell
app/pages/sections/order-items/catalogue-solutions/delete/confirmation/controller.test.js
   4:24  error  Useless path segments for "../../.../../../../../../logger", should be "../../../../../../logger"  import/no-useless-path-segments
  25:13  error  'orderItemId' is assigned a value but never used                                                   no-unused-vars
  29:26  error  'orderId' is not defined                                                                           no-undef
  36:64  error  'orderId' is not defined 
```

File: app/pages/sections/order-items/catalogue-solutions/delete/confirmation/manifest.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+    "title": "{{solutionName}} deleted from {{orderId}}",
+    "description": "This Catalogue Solution has been deleted from all the Service Recipients in your order along with any of its Additional Services.",
+    "continueButtonText": "Continue"
+}
```
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/delete/confirmation/manifest.json:5]: The file should end with a trailing new line.

File: app/pages/sections/order-items/catalogue-solutions/delete/contextCreator.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+import { addParamsToManifest } from '../../../../../helpers/contextCreators/addParamsToManifest';
+
+export const getContext = ({ orderId, orderItemId, solutionName, orderDescription }) => {
+  const context = ({
+    ...addParamsToManifest(manifest, {orderId, orderItemId, solutionName }),
+    orderDescription,
+    backLinkHref: `${baseUrl}/organisation/${orderId}/catalogue-solutions/${orderItemId}`,
+  });
+  return context;
+};
```
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/delete/contextCreator.js:12]: There are some linting issues in this file:

```shell
app/pages/sections/order-items/catalogue-solutions/delete/contextCreator.js
  5:28  error  Expected a line break after this opening brace   object-curly-newline
  5:83  error  Expected a line break before this closing brace  object-curly-newline
  7:38  error  A space is required after '{'                    object-curly-spacing
```

File: app/pages/sections/order-items/catalogue-solutions/delete/contextCreator.test.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      expect(context.noButtonText).toEqual(manifest.noButtonText);
+      expect(context.backLinkHref).toEqual(`${baseUrl}/organisation/${orderId}/catalogue-solutions/${orderItemId}`);
+    });
+
+    it('should return the yes button', () => {
+      const context = getContext({});
+      expect(context.yesButtonText).toEqual(manifest.yesButtonText);
+    });
+  });
+});
```
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/delete/contextCreator.test.js:52]: There are a couple of linting issues in this file:

```shell
app\pages\sections\order-items\catalogue-solutions\delete\contextCreator.test.js
  15:102  error  'orderItemId' is not defined  no-undef
  44:102  error  'orderItemId' is not defined  no-undef
```

File: app/pages/sections/order-items/catalogue-solutions/delete/controller.js [hunk 8]
```diff
@@ -1,5 +1,26 @@
...
+  const { orderItemId } = req.params;
+  const { solutionName } = req.params;
+  const orderDescription = await getOrderDescription({
+    req,
+    sessionManager,
+    accessToken,
+    logger,
+  });
+
+  return getContext({ orderId, orderItemId, solutionName, orderDescription });
```
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/delete/controller.js:21]: It looks like there may be a couple of linting issues here:

```shell
app/pages/sections/order-items/catalogue-solutions/delete/controller.js
  21:21  error  Expected a line break after this opening brace   object-curly-newline
  21:76  error  Expected a line break before this closing brace  object-curly-newline
```

File: app/pages/sections/order-items/catalogue-solutions/delete/manifest.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+    "description": "This Catalogue Solution will be deleted from all the Service Recipients in your order along with any of its Additional Services.",
+    "orderDescriptionTitle": "Order description",
+    "noButton": {
+      "text": "Cancel delete"
+    },
+    "yesButton": {
+      "text": "Confirm delete"
+    }
+  }
+
```
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/delete/manifest.json:13]: I think there is some surplus white space at the end of the file here. There should just be a single trailing new line.

File: app/pages/sections/order-items/catalogue-solutions/delete/routes.js [hunk 10]
```diff
@@ -1,29 +1,69 @@
...
+    });
+
+    logger.info(`navigating to catalogue ${solutionName} delete-catalogue-confirmation page`);
+    res.render('pages/sections/order-items/catalogue-solutions/delete/confirmation/template.njk', addContext({ context, user: req.user, csrfToken: req.csrfToken() }));
+  }));
+
+  router.post('/:orderItemId/confirmation/:solutionName/continue', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
+    const { orderId } = req.params;
+    
+    return res.redirect(`${config.baseUrl}/organisation/${orderId}/catalogue-solutions`);
```
Comment (bjss-chris) [inline app/pages/sections/order-items/catalogue-solutions/delete/routes.js:66]: There appears to be a number of linting issues in this file:

```shell
app/pages/sections/order-items/catalogue-solutions/delete/routes.js
   3:51  error  'app/pages/sections/order-items/catalogue-solutions/delete/controller.js' imported multiple times  import/no-duplicates
   6:41  error  'app/pages/sections/order-items/catalogue-solutions/delete/controller.js' imported multiple times  import/no-duplicates
   9:3   error  'formatFormData' is defined but never used                                                                             no-unused-vars
  10:3   error  'getOrderItemContext' is defined but never used                                                                        no-unused-vars
  11:3   error  'getOrderItemErrorContext' is defined but never used                                                                   no-unused-vars
  12:3   error  'getPageData' is defined but never used                                                                                no-unused-vars
  13:3   error  'setEstimationPeriod' is defined but never used                                                                        no-unused-vars
  15:10  error  'validateOrderItemFormBulk' is defined but never used                                                                  no-unused-vars
  16:10  error  'saveOrderItemBulk' is defined but never used                                                                          no-unused-vars
  17:10  error  'sessionKeys' is defined but never used                                                                                no-unused-vars
  18:10  error  'transformApiValidationResponse' is defined but never used                                                             no-unused-vars
  24:35  error  'solutionName' is assigned a value but never used                                                                      no-unused-vars
  49:13  error  'orderId' is assigned a value but never used                                                                           no-unused-vars
  49:21  error  Multiple spaces found before 'orderItemId'                                                                             no-multi-spaces
  49:23  error  'orderItemId' is assigned a value but never used                                                                       no-unused-vars
  49:48  error  A space is required before '}'                                                                                         object-curly-spacing
  65:1   error  Trailing spaces not allowed                                                                                            no-trailing-spaces
```

**Review by bjss-chris — CHANGES_REQUESTED** [review]
There appears to be a number of linting issues across quite a few files and there are some tests failing too.

Can you review the linting issues and failing tests? You'll also need to bring your branch up-to-date with the latest version of the development branch.

I'll re-review once you've fixed the linting issues and failing tests.

File: app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.js (hunk 1 above)
Comment (bjss-andrew) [reply app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.js:10]: how are these getting commited to a PR if the linter fails?

Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.js]: resolved

File: app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.test.js (hunk 2 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/delete/confirmation/contextCreator.test.js:23]: resolved

File: app/pages/sections/order-items/catalogue-solutions/delete/confirmation/controller.test.js (hunk 4 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/delete/confirmation/controller.test.js:47]: resolved

File: app/pages/sections/order-items/catalogue-solutions/delete/confirmation/controller.js (hunk 3 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/delete/confirmation/controller.js:10]: resolved

File: app/pages/sections/order-items/catalogue-solutions/delete/confirmation/manifest.json (hunk 5 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/delete/confirmation/manifest.json]: resolved

File: app/pages/sections/order-items/catalogue-solutions/delete/contextCreator.js (hunk 6 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/delete/contextCreator.js:14]: resolved

File: app/pages/sections/order-items/catalogue-solutions/delete/contextCreator.test.js (hunk 7 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/delete/contextCreator.test.js:53]: resolved

File: app/pages/sections/order-items/catalogue-solutions/delete/controller.js (hunk 8 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/delete/controller.js]: resolved

File: app/pages/sections/order-items/catalogue-solutions/delete/manifest.json (hunk 9 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/delete/manifest.json:14]: resolved

File: app/pages/sections/order-items/catalogue-solutions/delete/routes.js (hunk 10 above)
Comment (kiran-mastek) [reply app/pages/sections/order-items/catalogue-solutions/delete/routes.js:53]: resolved

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**